partitions_test.go

```diff
@@ -223,3 +225,14 @@ func Test_getAllMountedDevices(t *testing.T) {
 	sort.Sort(actual)
 	assert.Equal(t, actual, sort.StringSlice(expected))
 }
+
+func TestMaybeResolveLink(t *testing.T) {
+	// If path is not a symlink, the original path should be returned.
+	tmp, err := ioutil.TempDir("", "maybeResolveLink")
+	require.Nil(t, err)
+	resolvedPath := maybeResolveLink(tmp)
+	assert.Equal(t, tmp, resolvedPath)
+
+	resolvedPath = maybeResolveLink("/dev/disk/by-partuuid")
```

Comment:
Sham test. Question is: Is it necessary to test this? And if yes, I'm not entirely sure as to how this should be done robustly.

Reply:
I suppose you can make a link and a non-link just to be sure. Right now it appears to be testing just two non-links.

But you're right that this won't test the heart of the problem, which is the complex interaction between PARTUUID, Mender's partition indexes, and the bootloader's partition indexes. This needs to be tested at the acceptance test level. I looked into this when the PARTUUID feature was originally merged, but we need to build a specific image to test this because it must be based on using PARTUUID. It would be good to have such a test, but I'm also ok with merging without it, since it is quite a lot of work, and PARTUUID was a contributed feature.